gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -1472,30 +1472,34 @@ private Credential getCredential(
       throws IOException, GeneralSecurityException {
     Credential credential = null;
 
+    // In case of impersonating service account, service account scopes need to be expanded in order
+    // to request access token from IAM.
+    ImmutableList<String> scopes =
```

Comment:
Will it be simpler to just make scopes configurable and let users specify scopes that they need?

Reply:
I thought about that too, however I feel like this introduces another layer of complexity for user, it maybe only worth it until we have more use cases other than impersonation that need different scopes.